src/webServer/webServer.ts

```diff
@@ -137,6 +139,22 @@ namespace ts.server {
             /* eslint-enable no-restricted-globals */
 
             require: () => ({ module: undefined, error: new Error("Not implemented") }),
+            importServicePlugin: async (root: string, moduleName: string) => {
+                const packageRoot = combinePaths(root, "node_modules", moduleName);
+
+                const packageJsonResponse = await fetch(combinePaths(packageRoot, "package.json"));
+                const packageJson = await packageJsonResponse.json();
+                const browser = packageJson.browser;
+                if (!browser) {
+                    throw new Error("Could not load plugin. No 'browser' field found in package.json.");
+                }
+
+                const scriptPath = combinePaths(packageRoot, browser);
+
+                // TODO: TS rewrites `import(...)` to `require`. Use eval to bypass this
+                // eslint-disable-next-line no-eval
+                return (await eval(`import(${JSON.stringify(scriptPath)})`)).default;

```

Comment:
As mentioned in the call, I'm concerned that using `eval` could cause issues with CSP.

Reply:
I will need the TS team's help on this one. My current workaround is not intended to be what we ship but I don't know a better way to fix the issue right now

This is actually the second time I've run into this problem :) https://github.com/microsoft/vscode/blob/a171bcb8eb2424750a57823d87ac5156e15b03bd/src/vs/workbench/contrib/notebook/browser/view/renderers/webviewPreloads.ts#L2202

Could we perhaps add support for a special comment before the `import` that prevents TS from touching it on compile?

Reply:
Would [this suggestion](https://github.com/microsoft/TypeScript/pull/47377#discussion_r821033128) be helpful? It would push the `import` call syntax to the consumer of `createWebSystem`, who can compile using a compatible target that *doesn't* transform `import` to `require`.

Reply:
Oh yeah that seems promising! Sorry I didn't immediately understand why you proposed this in the original comment but will give it a try

Reply:
@rbuckton Quick question about this: I think I need to update `src/webServer/tsconfig.json` to have `"module": "ESNext"` or similar. However if I make this change, I get the error:

```
 Only 'amd' and 'system' modules are supported alongside --outFile.
```

I'm not very familiar with TS's project structure or how it uses project references. Can you suggest how to workaround this?

Reply:
I'll look, but I'm not sure there is a workaround. TS uses `--outFile` to bundle the relevant parts of the compiler into a single file. In this case, the resulting `webServer.js` contains the entire TypeScript compiler/language service. Changing `webServer` to be a module would require it to load the language service as an import, and I'm not sure what the impact of that change would be currently. 

Have you considered my suggestion here: https://github.com/microsoft/TypeScript/pull/47377#discussion_r821033128 ?

Rather than changing `webServer` to be a module, you could have the code that loads tsserver call `createWebSystem` with a callback that does the async import, and then consume it like this:

```ts
import ts from "typescript/lib/tsserver";

const sys = ts.createWebSystem(host, args, getExecutingFilePath, id => import(id)); // <- pass in the import
```

Reply:
> [...] the resulting `webServer.js` [...]

Sorry, I misspoke. `webServer` is bundled in `tsserver.js`, so there would be no way to load this as an ECMAScript module separately.

Reply:
@rbuckton Do you have any suggestions for alternative approaches?  Personally I'd really like to be able to add a comment that tells TS to not transpile the following `import` expression, but that's something that would need to be added to the compiler

Reply:
> [...] Personally I'd really like to be able to add a comment that tells TS to not transpile the following `import` expression, but that's something that would need to be added to the compiler

I'm not a fan of this approach, and I'm not sure its something we would do. Comments shouldn't affect emit of runtime code.

Is there a reason why passing an `id => import(id)` callback wouldn't be a viable option in this case? I think its perfectly reasonable to have a host environment tell `tsserver` how it can access resources in that environment, which is ostensibly what `ceateWebSystem` does.

If it isn't then we would likely need to completely rearchitect how `webServer` works, essentially extracting it out of `tsserver` into a separate project that itself *consumes* `tsserver.js`, which I think might be overkill. Other approaches would depend on some type of post-compile rewrite of `tsserver.js`, though I'm not much of a fan of that approach either.

Reply:
Ah, I better understand the issue. I've filed #48702 regarding changing our emit behavior for `import()`.

Reply:
This may be addressed by `--module node12`, though I'm not sure if that's compatible with `--outFile` either.

Reply:
So `--module node12` does allow native `import()`, but unfortunately is not compatible with `--outFile`. I've amended #48702 to specifically focus on using `import()` with `--module None`.

Once this has been addressed, what we would need to do here is add an additional `tsserverweb` project that mirrors the `tsserver` project, but uses `--target es2020` so that we could emit a native `import()` call.

Reply:
Or possibly change `tsserver.js` to `--target es2020`, though I'm not sure if that would negatively impact any existing consumers.

Reply:
Which one do you think sounds most promising? The post-compile one seems easier to me, both for implementation and as a consumer but it is also definitely a hack

I will need your help with either approach as well

Reply:
My only concern with a post-compile step is whether there are any other consumers of `tsserver.js` that use it in an environment where native `import()` isn't supported. The `tsserver.bootstrap.js` approach avoids that. 

@DanielRosenwasser, are you aware of any `tsserver.js` consumers that might run into an issue if that file contains dynamic import?

Reply:
I've considered this a bit more. I may be able to shoehorn it in in a different way by having a `tsserver.web.js` file that just contains this:

```js
var ts = ts || {};
ts.dynamicImport = id => import(id);
```

And adding a line to `webServer.ts` like this:

```ts
namespace ts {
  ...
  /* @internal */
  export declare var dynamicImport: ((id: string) => Promise<unknown>) | undefined;
  if (typeof importScripts === "function") {
    try {
      // if we're in a worker, attempt to load ts.dynamicImport
      importScripts("tsserver.web.js");
    }
    catch {
      // ignored
    }
  }
  ...
}
```

Since `importScripts` runs synchronously, it will pause execution of `tsserver.js` until the script is loaded. If the worker somehow doesn't support `import()` then the call to `importScripts` will throw (hence the `try..catch`), but the server will continue loading.

Reply:
I've made the changes necessary for this in 85e0fcad540d97609b862c25db7fc7bec340a8e7